### DISCUSSION TITLE
[Share] 打开obsidian笔记

### DISCUSSION THREAD

**Discussion by appwcn** [windingwind/zotero-actions-tags#335] [Action Scripts]
### Description

打开obsidian笔记
- 如果使用zotero integration插件，则Output Path定义为`{{itemKey}}.md`
- 如果使用zotlit插件，则Note filename定义为`<%= it.key %>.md`
- 如果使用Zotero Better Notes插件，则使用第二个代码

### Event

None

### Operation

条目菜单中

### Data

```javascript
/**
 * Open in Obsidian
 * @appw
 * @link https://github.com/windingwind/zotero-actions-tags/discussions/335
 */
const window = require("window");
if (!items.length) {
    return "所选项目不是Item";
}
// 使用Zotero Integration插件时，Output Path定义为{{itemKey}}.md
// 使用Zotlit插件时，Note filename定义为<%= it.key %>.md
let itemKey = items[0].key;
let urlob = `obsidian://open?file=${itemKey}`;
Zotero.launchURL(urlob);
```

```javascript
/**
 * Open in Obsidian
 * @link https://github.com/windingwind/zotero-actions-tags/discussions/335
 */
const window = require("window");
if (!items || items.length === 0 || !items[0].isNote()) {
    return "";
}
const itemID = items[0].id;
const syncStatus = Zotero.BetterNotes.api.sync.getSyncStatus(itemID);
if (syncStatus?.itemID != itemID) {
    return "Not synced note item";
}
const path = PathUtils.join(syncStatus.path, syncStatus.filename);
let urlob = `obsidian://open?path=${encodeURIComponent(path)}`;
Zotero.launchURL(urlob);
```

**Comment by XoYaPeng** [comment]
你好，如果用的是zotero better notes同步的笔记怎么办呢？

**Reply by windingwind** [reply]
可按需修改，未做测试

```js
if (!items || items.length === 0 || !items[0].isNote()) {
    return "";
}
const itemID = items[0].id;
const syncStatus = Zotero.BetterNotes.api.sync.getSyncStatus(itemID);
if (syncStatus?.itemID != itemID) {
    return "Not synced note item";
}
const path = PathUtils.join(syncStatus.path, syncStatus.filename);
let urlob = `obsidian://open?path=${encodeURIComponent(path)}`;
Zotero.launchURL(urlob);
```

**Reply by XoYaPeng** [reply]
感谢大佬解答我的问题，我还有疑问就是，上面说的(使用Item.key命名文件），意思是 ob 里的笔记名字要用 item.key 吗？如果是这样子，我怎么设置呢？我现在的文件名称都是doi -*******，doi 号是我自己放在笔记顶头，后面一串*号应该是 key 吧？我不太懂，想问问大佬。设置 doi 号也纯粹是为了笔记好找😂😂

**Reply by XoYaPeng** [reply]
我明天再贴个图出来，大佬帮我看下，感谢大佬(^🙏^)

**Reply by XoYaPeng** [reply]
`${(noteItem.getNoteTitle ? noteItem.getNoteTitle().replace(/[/\\?%*:|"<> ]/g, "-") : "")}${noteItem.key}.md`
大佬，帮我看下这个文件名能用不，我试了下，好像不行。找不到条目。下图是我ob中的文件名称。
![image](https://github.com/windingwind/zotero-actions-tags/assets/130196652/e9799975-8a37-41e9-8fbf-8745d7c53565)

**Reply by appwcn** [reply]
> 可按需修改，未做测试
> 
> ```js
> if (!items || items.length === 0 || !items[0].isNote()) {
>     return "Not note item";
> }
> const itemID = items[0].idl;
> const syncStatus = Zotero.BetterNotes.api.sync.getSyncStatus(itemID);
> if (syncStatus?.itemID != itemID) {
>     return "Not synced note item";
> }
> const path = PathUtils.join(syncStatus.path, syncStatus.filename);
> let urlob = encodeURIComponent(`obsidian://open?path=${path}`);
> Zotero.launchURL(urlob);
> ```

收到，这几天太忙了。等有空把这段代码放进去~ 谢谢大大

**Reply by XoYaPeng** [reply]
@appwcn 你尝试了之后能用不，我的貌似还不行😄等你好结果哈哈哈

**Reply by appwcn** [reply]
> @appwcn 你尝试了之后能用不，我的貌似还不行😄等你好结果哈哈哈

我没装better notes（插件开的很少，电脑性能带不动），大佬您可能要自己测试一下哈。弄好了告诉我，我可以补充进原始代码，分享给大家。

**Reply by XoYaPeng** [reply]
> > @appwcn 你尝试了之后能用不，我的貌似还不行😄等你好结果哈哈哈
> 
> 我没装better notes（插件开的很少，电脑性能带不动），大佬您可能要自己测试一下哈。弄好了告诉我，我可以补充进原始代码，分享给大家。

我试了，打不开，也不知道我哪里出了问题😂😂

**Reply by appwcn** [reply]
> > > @appwcn 你尝试了之后能用不，我的貌似还不行😄等你好结果哈哈哈
> > 
> > 
> > 我没装better notes（插件开的很少，电脑性能带不动），大佬您可能要自己测试一下哈。弄好了告诉我，我可以补充进原始代码，分享给大家。
> 
> 我试了，打不开，也不知道我哪里出了问题😂😂

那您再等等[windingwind](https://github.com/windingwind)大佬有空来帮你看看

**Reply by XoYaPeng** [reply]
嗯嗯好嘞

**Reply by windingwind** [reply]
谢谢反馈，刚看了下，之前有笔误，改动在原回复了。需要是已通过BN同步的笔记方可通过该动作打开md。

**Reply by XoYaPeng** [reply]
> 谢谢反馈，刚看了下，之前有笔误，改动在原回复了。需要是已通过BN同步的笔记方可通过该动作打开md。

太强了大佬，感谢感谢!!!!这一来也就更方便知道哪篇笔记在Obsidian里了！🙏🙏
